apps/src/blockly/blockSvgUnused.js

```diff
@@ -0,0 +1,213 @@
+import msg from '@cdo/locale';
+const FRAME_MARGIN_SIDE = 15;
+const FRAME_MARGIN_TOP = 10;
+const FRAME_MARGIN_BOTTOM = 5;
+
+const FRAME_HEADER_HEIGHT = 25;
+
+export const BlockSvgUnused = function(block, helpClickFunc) {
+  this.block_ = block;
+  this.helpClickFunc_ = helpClickFunc;
+
+  this.frameGroup_ = undefined;
+  this.frameClipRect_ = undefined;
+  this.frameBase_ = undefined;
+  this.frameHeader_ = undefined;
+  this.frameText_ = undefined;
+  this.frameHelp_ = undefined;
+
+  this.initChildren();
+};
+
+BlockSvgUnused.UNUSED_BLOCK_HELP_EVENT = 'blocklyUnusedBlockHelpClicked';
+
+BlockSvgUnused.prototype.initChildren = function() {
+  this.frameGroup_ = Blockly.utils.dom.createSvgElement('g', {
+    class: 'blocklyUnusedFrame'
+  });
+
+  var clip = Blockly.utils.dom.createSvgElement(
+    'clipPath',
+    {
+      id: `frameClip${this.block_.id}`
+    },
+    this.frameGroup_
+  );
+  this.frameClipRect_ = Blockly.utils.dom.createSvgElement(
+    'rect',
+    {
+      x: -FRAME_MARGIN_SIDE,
+      y: -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT),
+      height: FRAME_HEADER_HEIGHT
+    },
+    clip
+  );
+
+  this.frameBase_ = Blockly.utils.dom.createSvgElement(
+    'rect',
+    {
+      x: -FRAME_MARGIN_SIDE,
+      y: -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT),
+      fill: '#e7e8ea',
+      stroke: '#c6cacd',
+      rx: 15,
+      ry: 15
+    },
+    this.frameGroup_
+  );
+
+  this.frameHeader_ = Blockly.utils.dom.createSvgElement(
+    'rect',
+    {
+      x: -FRAME_MARGIN_SIDE,
+      y: -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT),
+      fill: '#c6cacd',
+      rx: 15,
+      ry: 15,
+      'clip-path': `url(#frameClip${this.block_.id.replace(/\)/g, '\\)')})`
+    },
+    this.frameGroup_
+  );
+
+  var frameTextVerticalPosition = -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT / 2);
+  if (Blockly.ieVersion()) {
+    // in non-IE browsers, we use dominant-baseline to vertically center
+    // our text. That is unfortunately not supported in IE, so we
+    // manually offset by 4 pixels to compensate.
+    frameTextVerticalPosition += 4;
+  }
+  this.frameText_ = Blockly.utils.dom.createSvgElement(
+    'text',
+    {
+      class: 'blocklyText',
+      style: 'font-size: 12pt;fill: black',
+      y: frameTextVerticalPosition,
+      'dominant-baseline': 'central'
+    },
+    this.frameGroup_
+  );
+  this.frameText_.appendChild(document.createTextNode(msg.unusedCode()));
+
+  this.frameHelp_ = Blockly.utils.dom.createSvgElement(
+    'g',
+    {
+      class: 'blocklyHelp'
+    },
+    this.frameGroup_
+  );
+  Blockly.utils.dom.createSvgElement(
+    'circle',
+    {
+      fill: '#7665a0',
+      r: FRAME_HEADER_HEIGHT * 0.75 * 0.5
+    },
+    this.frameHelp_
+  );
+  Blockly.utils.dom
+    .createSvgElement(
+      'text',
+      {
+        class: 'blocklyText',
+        y: Blockly.ieVersion() ? 4 : 0, // again, offset text manually in IE
+        'dominant-baseline': 'central',
```

Comment:
add `dominant-baseline` and `text-anchor` styles. These appeared to be present on the element using CDO Blockly, but weren't set explicitly here, so my guess is they were applied using CSS at some other layer

Reply:
(Chrome inspector is useful for sourcing CSS on an element.)

Reply:
This comes from in CDO Blockly https://github.com/code-dot-org/blockly/blob/main/core/ui/css.js#L162